warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/CompositeIngest.java

```diff
@@ -379,17 +412,16 @@ private void compilePatterns() {
         // Create the composite field from the event Fields members
         public void addCompositeFields(List<NormalizedContentInterface> compositeFields, Multimap<String,NormalizedContentInterface> eventFields,
                         String compositeFieldName, String replacement, String[] grouping, GroupingPolicy groupingPolicy, boolean allowMissing, String[] fields,
-                        int pos, String startSeparator, String endSeparator, StringBuilder originalValue, StringBuilder normalizedValue,
-                        Map<String,String> markings) {
-            String separator = "";
+                        int pos, StringBuilder originalValue, StringBuilder normalizedValue, Map<String,String> markings, boolean isOverloadedField) {
+            String separator = (pos > 0) ? COMPOSITE_DEFAULT_SEPARATOR : "";
             // append any constants that have been specified
             while (pos < fields.length && isConstant(fields[pos])) {
                 String constant = getConstant(fields[pos++]);
-                originalValue.append(constant);
+                if (!isOverloadedField || (isOverloadedField && pos == 0))
```

Comment:
... then pos cannot be zero here. or did i miss something?

Reply:
Yup, this needs to be changed.

Reply:
Fixed!